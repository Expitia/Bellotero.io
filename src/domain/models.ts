export interface IOption {
  text: string
  route: string
}

export interface IReview {
  name: string
  comment: string
  position: string
}

export enum Views {
  DEFAULT = '/',
  REVIEWS = '/page-1',
  CALCULATOR = '/page-2',
}

export interface IResponseMenu {
  menu: {
    items: IOption[]
  }
}

export interface IResponseSlider {
  slider: {
    title: string
    reviews: IReview[]
  }
}

export interface IResponseCalculator {
  calculator: {
    title: string
    description: string
  }
}
