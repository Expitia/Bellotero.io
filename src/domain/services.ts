import axios from 'axios'
import { IResponseCalculator, IResponseMenu, IResponseSlider } from './models'

const serviceURL =
  'https://raw.githubusercontent.com/Bernabe-Felix/Bellotero/master'

const executeRequest = async options => {
  return axios
    .request(options)
    .then(response => response.data)
    .catch(error => error)
}

export const getOptions = (): Promise<IResponseMenu | Error> =>
  executeRequest({
    method: 'GET',
    url: `${serviceURL}/app.json`,
  })

export const getReviews = (): Promise<IResponseSlider | Error> =>
  executeRequest({
    method: 'GET',
    url: `${serviceURL}/page1.json`,
  })

export const getCalculatorInfo = (): Promise<IResponseCalculator | Error> =>
  executeRequest({
    method: 'GET',
    url: `${serviceURL}/page2.json`,
  })
