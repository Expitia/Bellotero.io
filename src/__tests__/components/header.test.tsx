import { shallow } from 'enzyme'
import React from 'react'
import Header from '../../components/header'
import { Views } from '../../domain/models'

describe('Header', () => {
  it('should render Header component', () => {
    const component = shallow(
      <Header
        options={[
          {
            text: 'Calculator example',
            route: Views.CALCULATOR,
          },
        ]}
        onSelectOption={() => undefined}
        selectedOption={Views.CALCULATOR}
      />,
    )

    expect(component).toMatchSnapshot()
  })
})
