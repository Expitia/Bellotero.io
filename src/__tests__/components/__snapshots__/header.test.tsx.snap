// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render Header component 1`] = `
<Navbar
  color="white"
  expand="md"
  light={true}
  tag="nav"
>
  <div
    className="container d-flex w-100"
  >
    <NavbarBrand
      href="/"
      tag="a"
    >
      <img
        src="bellotero.png"
      />
    </NavbarBrand>
    <NavbarToggler
      color="red"
      onClick={[Function]}
      tag="button"
      type="button"
    />
    <Collapse
      appear={false}
      enter={true}
      exit={true}
      in={false}
      isOpen={false}
      mountOnEnter={false}
      navbar={true}
      onEnter={[Function]}
      onEntered={[Function]}
      onEntering={[Function]}
      onExit={[Function]}
      onExited={[Function]}
      onExiting={[Function]}
      tag="div"
      timeout={350}
      unmountOnExit={false}
    >
      <Nav
        className="ml-auto"
        navbar={true}
        tag="ul"
        vertical={false}
      >
        <NavItem
          key="0"
          tag="li"
        >
          <NavLink
            className="nav-link"
            onClick={[Function]}
            tag="a"
          >
            Calculator example
          </NavLink>
        </NavItem>
      </Nav>
    </Collapse>
  </div>
</Navbar>
`;
